llvm/include/llvm/CodeGen/GlobalISel/GIMatchTableExecutor.h

```diff
@@ -713,6 +731,12 @@ class GIMatchTableExecutor {
     return Ret;
   }
 
+  static ArrayRef<MachineOperand> getVariadicOperands(const MachineInstr &MI,
```

Comment:
Rename to something more generic, given that the number of operands to skip. Alternatively infer from the instruction properties where the variadic arguments begin

Reply:
Will do, this can probably go into a separate file as well with other Combiner utils.

The direction I took with GIVariadic is that it's just a "variable operand pack", so 1 or more operands bundled together. It's not tied to the nature of the instruction itself.
You could write `G_MUL $x, GIVariadic<>:$lhsAndRhs` and that would also work.

Writing that now I realize that maybe this should be called something else, maybe something like a operand "bundle" or "pack". Or I should revert that decision and make it all about variadic instructions?

I like the idea of making them neutral because it gives more room to breathe for future feature, for instance if we have some kind of templated pattern for multiple opcodes each with a different number of inputs, we could use a GIVariadic to model the operands.